PhysicsTools/NanoAOD/plugins/MuonFSRProducer.cc

```diff
@@ -0,0 +1,247 @@
+// system include files
```

Comment:
This producer is of much more general usage than for nanoAOD, it should not go here.
I would suggest you to open a separate PR to upstream CMSSW to have it integrated in a muon software area, and have it reviewed (including the recently introduced requirements on code formatting and quality).